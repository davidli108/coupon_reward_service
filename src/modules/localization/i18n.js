import i18n from 'i18next';
import parseDomain from 'parse-domain';
import * as R from 'ramda';
import { initReactI18next } from 'react-i18next';

import translationDE from './locales/de/translation';
import translationEN from './locales/en/translation';
import translationFR from './locales/fr/translation';
import translationGB from './locales/gb/translation';

const resources = {
  de: {
    translation: translationDE,
  },
  en: {
    translation: translationEN,
  },
  fr: {
    translation: translationFR,
  },
  gb: {
    translation: translationGB,
  },
};

const Locale = {
  en: 'en',
  de: 'de',
  fr: 'fr',
  gb: 'gb',
};

const localeToDomain = {
  com: Locale.en,
  de: Locale.de,
  fr: Locale.fr,
  'co.uk': Locale.gb,
};

const fallbackLocale = Locale.en;

export const getDomainAttrs = () => {
  const domain = window.location.hostname;
  const protocol = window.location.protocol;

  const parsedDomain = parseDomain(domain);
  const tld = R.propOr(null, 'tld', parsedDomain);
  const enDomain = domain.replace(tld, 'com');
  const enOrigin = `${protocol}//${enDomain}`;

  return {
    domain,
    protocol,
    tld,
    enDomain,
    enOrigin,
  };
};

export const redirectToEnOrigin = () => {
  const domainAttrs = getDomainAttrs();
  window.location.href = domainAttrs.enOrigin;
};

export const getLocale = () => {
  const domainAttrs = getDomainAttrs();
  return localeToDomain[domainAttrs.tld] || fallbackLocale;
};

export const getLocaleConfig = () => {
  return {
    // TODO: enable authentication on all domains after feature support added
    isAuthenticationAvailable: true,
    // TODO: enable follow store on all domains after feature support added
    isFollowStoreAvailable: true,
  };
};

export const initializeI18n = () => {
  // eslint-disable-next-line import/no-named-as-default-member
  i18n.use(initReactI18next).init({
    resources,
    lng: getLocale(),
    fallbackLng: fallbackLocale,
    keySeparator: '.',
    interpolation: {
      escapeValue: false,
    },
  });
};

export default i18n;
