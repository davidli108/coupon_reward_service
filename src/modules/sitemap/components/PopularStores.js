// @flow
import React, { useState, useEffect } from 'react';
import { withTranslation } from 'react-i18next';
import styled from 'styled-components';
import breakpoint from 'styled-components-breakpoint';
import placeholder from '@modules/coupons/assets/image-placeholder.png';
import { MdChevronRight } from 'react-icons/md';
import popular_stores from '../assets/popular_stores';
import { NavLink } from 'react-router-dom';
import Slider from 'react-slick';
import { slider_settings } from '../constants';

type PopuparStoresProps = {
  t: Function,
  i18n: Object,
};

const PopularStores = ({ t, i18n }: PopuparStoresProps) => {
  const [popularStores, setPopularStores] = useState([]);

  useEffect(() => {
    setPopularStores(popular_stores[i18n.language]);
  }, [popular_stores, i18n.language]);

  return (
    <PopularStores.Wrapper>
      <Slider {...slider_settings}>
        {popularStores &&
          popularStores.map((store: any, key: number) => (
            <PopularStores.ItemWrapper key={key}>
              <PopularStores.Item>
                <PopularStores.Figure>
                  <img
                    src={store.logo}
                    onError={e => {
                      e.target.onerror = null;
                      e.target.src = placeholder;
                    }}
                    alt={store.name}
                  />
                </PopularStores.Figure>
                <b>{store.name}</b>
                <div>
                  <MdChevronRight />
                  <span>
                    {store.discount}{' '}
                    {store.kind === 'paid_placements'
                      ? t('global.cashBack')
                      : ''}
                  </span>
                </div>
                <NavLink to={`/coupons/${store.short_name}`}>
                  {t('sitemap.visit_store')}
                </NavLink>
              </PopularStores.Item>
            </PopularStores.ItemWrapper>
          ))}
      </Slider>
    </PopularStores.Wrapper>
  );
};

PopularStores.Wrapper = styled.div`
  display: block;
  padding: 0 15px 41px;
  margin: 0 0 43px;
  position: relative;
  border-bottom: 1px solid #f0f1f4;

  .slick-list {
    overflow: hidden;
  }

  .slick-slider {
    margin: 0 -15px;
  }

  .slick-slide {
    &:last-of-type {
      > div > div::after {
        display: none;
      }
    }
  }

  .slick-track {
    display: flex;

    *:focus {
      outline: none;
    }
  }

  .slick-arrow {
    display: flex !important;
    align-items: center;
    width: 77px;
    height: 77px;
    border-radius: 50%;
    box-shadow: 0px 14px 34px rgba(0, 0, 0, 0.2);

    &.slick-prev,
    &.slick-next {
      border: 0;
      padding: 0;
      position: absolute;
      z-index: 10;
      bottom: 50%;
      background-color: #fff;

      svg {
        width: 20px;
        height: 20px;
        display: block;
        color: #979797;
      }
    }

    &.slick-prev {
      left: -54px;
      justify-content: flex-end;

      svg {
        margin: 0 12px 0 0;
      }
    }

    &.slick-next {
      right: -54px;

      svg {
        margin: 0 0 0 12px;
      }
    }
  }

  ${breakpoint('sm')`
    .slick-arrow {
      &.slick-prev,
      &.slick-next {
        width: 97px;
        height: 97px;
      }

      &.slick-prev {
        left: -63px;

        svg {
          margin: 0 20px 0 0;
        }
      }

      &.slick-next {
        right: -63px;

        svg {
          margin: 0 0 0 20px;
        }
      }
    }
  `}
`;

PopularStores.List = styled.ul`
  display: flex;
  margin: 0 -15px;
`;

PopularStores.ItemWrapper = styled.div`
  padding: 0 15px;
  box-sizing: border-box;
  flex: 1;
  position: relative;

  ::after {
    content: '';
    display: block;
    position: absolute;
    top: 50%;
    left: 100%;
    transform: translate(0, -50%);
    height: 140px;
    width: 1px;
    background-image: linear-gradient(
      to bottom,
      #b3bbc2 20%,
      rgba(255, 255, 255, 0) 0%
    );
    background-position: left;
    background-size: 1px 10px;
    background-repeat: repeat-y;
  }
`;

PopularStores.Item = styled.div`
  padding: 43px 0 35px;
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
  position: relative;

  b {
    display: block;
    font-size: 16px;
    line-height: 19px;
    font-weight: 900;
    text-align: center;
    color: #6b7480;
    margin: 0 0 7px;
  }

  > div {
    margin: 0 0 22px;
    display: flex;
    align-items: center;

    span {
      color: #7ed321;
      font-size: 12px;
      font-weight: 900;
      letter-spacing: 0.46px;
    }
  }

  a {
    display: block;
    color: #374b5a;
    text-decoration: underline;
    font-size: 16px;
    line-height: 19px;
    font-weight: 600;
    text-align: center;
  }
`;

PopularStores.Figure = styled.figure`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 0 27px;
  padding: 0;
  width: 100px;
  height: 100px;
  border-radius: 15px;
  position: relative;

  img {
    display: block;
    max-width: 100%;
    object-fit: contain;
    position: relative;
    z-index: 1;
  }

  ::after,
  ::before {
    content: '';
    display: block;
    width: 11px;
    height: 11px;
    border-radius: 50%;
    background-color: #fff;
    position: absolute;
    top: 50%;
    left: -6px;
    transform: translate(0, -50%);
  }

  ::after {
    left: auto;
    right: -6px;
  }
`;

export default withTranslation()(PopularStores);
