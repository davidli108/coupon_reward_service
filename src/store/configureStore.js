// @flow
import { routerMiddleware } from 'connected-react-router';
import { createBrowserHistory, createMemoryHistory } from 'history';
import { applyMiddleware, compose, createStore } from 'redux';
import { persistStore } from 'redux-persist';
import createSagaMiddleware from 'redux-saga';
import thunk from 'redux-thunk';

import axiosMiddleware from './middlewares/axiosMiddleware';
import loggerMiddleware from './middlewares/loggerMiddleware';
import promiseMiddleware from './middlewares/promiseMiddleware';
import rootReducer from './rootReducer';
import rootSaga from './rootSaga';

export const isServer = !(
  typeof window !== 'undefined' &&
  window.document &&
  window.document.createElement
);

export default () => {
  // const history = createHistory();
  const history = isServer
    ? createMemoryHistory({ initialEntries: ['/'] })
    : createBrowserHistory();

  const initialState = !isServer ? window.__PRELOADED_STATE__ || {} : {};

  const sagaMiddleware = createSagaMiddleware();

  const middlewares = [
    routerMiddleware(history),
    axiosMiddleware,
    thunk,
    promiseMiddleware,
    sagaMiddleware,
  ];

  if (!isServer && typeof jest === 'undefined') {
    middlewares.push(loggerMiddleware);
  }

  const composeEnhancers = isServer
    ? compose
    : window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

  // $FlowFixMe
  const store = createStore(
    (rootReducer: Object)(history),
    initialState,
    composeEnhancers(applyMiddleware(...middlewares)),
  );
  const persistor = persistStore(store);

  sagaMiddleware.run(rootSaga);

  // $FlowFixMe
  if (module.hot) {
    // $FlowFixMe
    module.hot.accept('./rootReducer', () => {
      // eslint-disable-next-line global-require
      const nextRootReducer: Object = require('./rootReducer').default(history);
      store.replaceReducer(nextRootReducer);
    });
  }

  return { history, persistor, store };
};
