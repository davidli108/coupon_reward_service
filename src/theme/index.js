// @flow
import {
  GlobalStyle as BaseGlobalStyle,
  theme as baseTheme,
} from './themes/base';

const themeMap = {
  base: {
    theme: baseTheme,
    GlobalStyle: BaseGlobalStyle,
  },
};

export const getTheme = (themeName: string) => themeMap[themeName] || {};
