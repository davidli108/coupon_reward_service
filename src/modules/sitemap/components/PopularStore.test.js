import '../../../setupTests.js';
import React from 'react';
import PopularStores from './PopularStores';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from '../../../store/configureStore';
import { ThemeProvider } from 'styled-components';
import { getTheme } from '../../../theme';
import { initializeI18n } from '../../../modules/localization/i18n';
import { BrowserRouter as Router } from 'react-router-dom';
import 'jest-styled-components';

initializeI18n();

const { store } = configureStore();
const { theme } = getTheme('base');

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <ThemeProvider theme={theme}>
            <PopularStores
              isLoadingCategories={false}
              isLoadingStores={false}
            />
          </ThemeProvider>
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
