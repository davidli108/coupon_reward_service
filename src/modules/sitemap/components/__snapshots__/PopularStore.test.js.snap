// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: block;
  padding: 0 15px 41px;
  margin: 0 0 43px;
  position: relative;
  border-bottom: 1px solid #f0f1f4;
}

.c0 .slick-list {
  overflow: hidden;
}

.c0 .slick-slider {
  margin: 0 -15px;
}

.c0 .slick-slide:last-of-type > div > div::after {
  display: none;
}

.c0 .slick-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .slick-track *:focus {
  outline: none;
}

.c0 .slick-arrow {
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 77px;
  height: 77px;
  border-radius: 50%;
  box-shadow: 0px 14px 34px rgba(0,0,0,0.2);
}

.c0 .slick-arrow.slick-prev,
.c0 .slick-arrow.slick-next {
  border: 0;
  padding: 0;
  position: absolute;
  z-index: 10;
  bottom: 50%;
  background-color: #fff;
}

.c0 .slick-arrow.slick-prev svg,
.c0 .slick-arrow.slick-next svg {
  width: 20px;
  height: 20px;
  display: block;
  color: #979797;
}

.c0 .slick-arrow.slick-prev {
  left: -54px;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c0 .slick-arrow.slick-prev svg {
  margin: 0 12px 0 0;
}

.c0 .slick-arrow.slick-next {
  right: -54px;
}

.c0 .slick-arrow.slick-next svg {
  margin: 0 0 0 12px;
}

@media (min-width:36em) {
  .c0 .slick-arrow.slick-prev,
  .c0 .slick-arrow.slick-next {
    width: 97px;
    height: 97px;
  }

  .c0 .slick-arrow.slick-prev {
    left: -63px;
  }

  .c0 .slick-arrow.slick-prev svg {
    margin: 0 20px 0 0;
  }

  .c0 .slick-arrow.slick-next {
    right: -63px;
  }

  .c0 .slick-arrow.slick-next svg {
    margin: 0 0 0 20px;
  }
}

<div
  className="c0"
>
  <div
    className="slick-slider slick-initialized"
  >
    
    <div
      className="slick-list"
    >
      <div
        className="slick-track"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "WebkitTransform": "translate3d(0px, 0px, 0px)",
            "WebkitTransition": "-webkit-transform 500ms ease",
            "msTransform": "translateX(0px)",
            "opacity": 1,
            "transform": "translate3d(0px, 0px, 0px)",
            "transition": "transform 500ms ease",
          }
        }
      />
    </div>
    
    
  </div>
</div>
`;
